mlir/include/mlir/Dialect/LLVMIR/LLVMDialect.h

```diff
@@ -87,6 +88,13 @@ class GEPArg : public PointerUnion<Value, GEPConstantIndex> {
 } // namespace LLVM
 } // namespace mlir
 
+namespace mlir {
+namespace LLVM {
+struct AssumeAlignTag {};
+struct AssumeSeparateStorageTag {};
```

Comment:
Can you try to add them in an extraClassDeclaration in the tablegen definition of the assume intrinsic?

```
let extraClassDeclaration = [{
  struct AssumeAlignTag {};
  struct AssumeSeparateStorageTag {};
}];
```
Afterwards they should be accessible as AssumeOp::AlignTag / AssumeOp::SeparateStorageTag. 

I think that may work and would clearly link them with the op.

Reply:
I tried this and it did not work because tblgen puts `extraClassDeclaration` at the end of the class declaration, and the builder declarations could not find them:

```cpp
class AssumeOp {
  static AssumeOp build(AlignTag);  // <-- builders generated
  struct AlignTag {};  // <-- extraClassDeclaration is put after builders thus builders could not find them
};
```

Reply:
That's unfortunate. Then let's stick to your original solution!